indico/modules/events/editing/client/js/components/FileManager/util.js

```diff
@@ -0,0 +1,88 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import deleteFileURL from 'indico-url:files.delete_file';
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import * as actions from './actions';
+
+export const FileManagerContext = React.createContext(null);
+
+export const filePropTypes = {
+  filename: PropTypes.string.isRequired,
+  downloadURL: PropTypes.string.isRequired,
+  uuid: PropTypes.string.isRequired,
+  claimed: PropTypes.bool,
+  state: PropTypes.oneOf(['added', 'modified', 'deleted']),
+};
+
+async function uploadFile(file, url, onUploadProgress) {
+  const formData = new FormData();
+  formData.append('file', file);
+
+  try {
+    const {data} = await indicoAxios.post(url, formData, {
+      headers: {'content-type': 'multipart/form-data'},
+      onUploadProgress,
+    });
+    return data;
+  } catch (e) {
+    handleAxiosError(e, false, true);
```

Comment:
what happens to the calling code if it fails? does it go kaboom or just hide the failed upload?

Reply:
I believe it just kabooms 💣

Reply:
We would have to add an additional action to "finish" the upload once there is an error. I think it can be done in a future task as it's not urgent right now.

Reply:
yeah, agreed